enterprise/com/src/test/java/org/neo4j/com/storecopy/TransactionCommittingResponseUnpackerTest.java

```diff
@@ -156,16 +160,55 @@ public void panicAndSkipBatchOnApplyingFailure() throws Throwable
                 errorMessage, kernelHealth.getCauseOfPanic().getMessage() );
 
         // 2 transactions where committed by none was closed.
-        verify( txIdStore, times( 2 ) ).transactionCommitted( anyLong(), anyLong() );
+        verify( txIdStore, times( 2 ) ).transactionCommitted( anyLong(), anyLong(), anyLong() );
         verify( txIdStore, times( 2 ) ).transactionClosed( anyLong(), anyLong(), anyLong() );
     }
 
     /*
-     * Tests that shutting down the response unpacker while in the middle of committing a transaction will
-     * allow that transaction stream to complete committing. It also verifies that any subsequent transactions
-     * won't begin the commit process at all.
+     * Tests that we unfreeze active transactions after commit and after apply of batch if batch length (in time)
+     * is larger than safeZone time.
      */
     @Test
+    public void shouldUnfreezeKernelTransactionsAfterApplyIfBatchIsLarge() throws Throwable
+    {
+        // GIVEN
+        final TransactionAppender appender = mockedTransactionAppender();
+        final IndexUpdatesValidator indexUpdatesValidator = mockedIndexUpdatesValidator(  );
+        final KernelHealth kernelHealth = newKernelHealth();
+        final long idReuseSafeZoneTime = 100;
+
+        Dependencies deps = new MockedDependencies()
+                .indexUpdatesValidator( indexUpdatesValidator )
+                .kernelHealth( kernelHealth )
+                .transactionAppender( appender )
+                .idReuseSafeZoneTime( idReuseSafeZoneTime );
+
+        int maxBatchSize = 3;
+        TransactionCommittingResponseUnpacker unpacker = new TransactionCommittingResponseUnpacker( deps, maxBatchSize );
+        unpacker.start();
+
+        // WHEN
+        int txCount = maxBatchSize;
+        int doesNotMatter = 1;
+        unpacker.unpackResponse(
+                new DummyTransactionResponse( doesNotMatter, txCount, appender, maxBatchSize, idReuseSafeZoneTime + 1 ),
+                NO_OP_TX_HANDLER );
+
+        // THEN
+        KernelTransactions kernelTransactions = deps.kernelTransactions();
+        BatchingTransactionRepresentationStoreApplier applier =
+                deps.transactionRepresentationStoreApplier();
+        InOrder inOrder = inOrder( kernelTransactions, applier );
+        inOrder.verify( applier, times( 1 ) ).closeBatch();
+        inOrder.verify( kernelTransactions, times( 1 ) ).unblockNewTransactions();
+    }
```

Comment:
All tests in `TransactionCommittingResponseUnpackerTest` are also mock based. We have integration tests for this in `TerminationOfSlavesDuringPullUpdatesTest`.